src/core/src/service.rs

```diff
@@ -289,6 +351,126 @@ mod tests {
+        let mut svc = Poet2Service::new( Box::new(cal_summarize));
+        let mut summarize_block_res = svc.finalize_block();
+        assert_eq!(summarize_block_res, BlockId::from(vec![99,99,99]));
+    }
+
+    #[test]
+    fn test_cal_wait_time_min_lessthan_maxtime(){
+        let mut test_mock_str = HashMap::new();
+        let mut str1 = String::from("300");
+        let mut str2 = String::from("500");
+        test_mock_str.insert(String::from("sawtooth.consensus.min_wait_time"), str1.clone());
+        test_mock_str.insert(String::from("sawtooth.consensus.max_wait_time"), str2.clone());
+        let test_result: Result<HashMap<String, String>, Error> = Ok(test_mock_str.clone());
+     	let test_block_id = BlockId::from(vec![78,39]);
+        let blk: Block = Block{block_id: BlockId::from(vec![11,22]), previous_id: BlockId::from(vec![11,12]), signer_id: PeerId::from(vec![11,25]), block_num: 112, payload:vec![67,23], summary: vec![45,76] };
+        let mut test_block = HashMap::new();
+        test_block.insert(test_block_id, blk);
+
+        let cal_time = MockServiceUT::new(Ok(test_mock_str), Ok(vec![100,200]), Ok(BlockId::from(vec![78,39])), Ok(test_block));
+        cal_time.get_settings.return_value(test_result);
+        let mut svc = Poet2Service::new( Box::new(cal_time));
+        let mut wt = svc.calculate_wait_time(Default::default()).as_secs();
+        let mut min_time = str1.parse::<u64>().unwrap();
+        let mut max_time = str2.parse::<u64>().unwrap();
+        if wt > min_time && wt < max_time{
+            assert_gt!(wt, min_time);
+            assert_lt!(wt, max_time);
+            debug!("wait_time is generated within the range {:?} and {:?}, {:?}",min_time, max_time, wt);
+        }
+    }
+	
+    #[test]
+    fn test_cal_wait_time_null_values(){
+        let mut test_mock_str = HashMap::new();
+        let mut str1 = String::from(" ");
+        let mut str2 = String::from(" ");
+        test_mock_str.insert(String::from("sawtooth.consensus.min_wait_time"), str1.clone());
+        test_mock_str.insert(String::from("sawtooth.consensus.max_wait_time"), str2.clone());
+        let test_result: Result<HashMap<String, String>, Error> = Ok(test_mock_str.clone());
+     	let test_block_id = BlockId::from(vec![78,39]);
+        let blk: Block = Block{block_id: BlockId::from(vec![11,22]), previous_id: BlockId::from(vec![11,12]), signer_id: PeerId::from(vec![11,25]), block_num: 112, payload:vec![67,23], summary: vec![45,76] };
+        let mut test_block = HashMap::new();
+        test_block.insert(test_block_id, blk);
+
+        let cal_time = MockServiceUT::new(Ok(test_mock_str), Ok(vec![100,200]), Ok(BlockId::from(vec![78,39])), Ok(test_block));
+        cal_time.get_settings.return_value(test_result);
+        let mut svc = Poet2Service::new( Box::new(cal_time));
+        let mut wt = svc.calculate_wait_time(Default::default()).as_secs();
+        assert_eq!(wt, 0);

```

Comment:
Should this be 0 or some default value?

Reply:
Anything is fine.As default value is not declared in tests scope, made it to 0.

Reply:
Try having some other non-zero default value and then assert it.